packages/analytics-js-plugins/src/utilities/retryQueue/RetryQueue.ts

```diff
@@ -138,27 +138,15 @@ class RetryQueue implements IQueue<QueueItemData> {
     const batchOptions = options.batch as BatchOpts;
 
     this.batch.enabled = batchOptions.enabled === true;
-
-    if (checks.isDefined(batchOptions.maxSize)) {
-      this.batch.maxSize = +(batchOptions.maxSize as number) || DEFAULT_MAX_BATCH_SIZE_BYTES;
-    }
-
-    if (checks.isDefined(batchOptions.maxItems)) {
-      this.batch.maxItems = +(batchOptions.maxItems as number) || DEFAULT_MAX_BATCH_ITEMS;
-    }
-
-    if (checks.isDefined(batchOptions.flushInterval)) {
-      this.batch.flushInterval =
-        +(batchOptions.flushInterval as number) || DEFAULT_BATCH_FLUSH_INTERVAL_MS;
-    }
-
-    // Set upper cap on the batch size
     if (this.batch.enabled) {
-      let maxSize = this.batch?.maxSize ?? DEFAULT_MAX_BATCH_SIZE_BYTES;
-      if (maxSize > DEFAULT_MAX_BATCH_SIZE_BYTES) {
-        maxSize = DEFAULT_MAX_BATCH_SIZE_BYTES;
+      this.batch.maxSize = batchOptions.maxSize || DEFAULT_MAX_BATCH_SIZE_BYTES;
+      // Set upper cap on the batch size
+      if (this.batch.maxSize > DEFAULT_MAX_BATCH_SIZE_BYTES) {
```

Comment:
you can use Math.min(batchOptions.maxSize || DEFAULT_MAX_BATCH_SIZE_BYTES, DEFAULT_MAX_BATCH_SIZE_BYTES)

Reply:
Thanks. Done.